### PR TITLE
Disable asset transfers on Statemint (against Statemint release branch v6.0.1)

### PR DESCRIPTION
**Pull request by KiChjang**
#834

### CODE REVIEW

File: polkadot-parachains/statemine/src/lib.rs [hunk 1]
```diff
@@ -584,8 +584,8 @@ impl pallet_xcm::Config for Runtime {
 	// ... but disallow generic XCM execution. As a result only teleports and reserve transfers are allowed.
 	type XcmExecuteFilter = Nothing;
 	type XcmExecutor = XcmExecutor<XcmConfig>;
-	type XcmTeleportFilter = Everything;
-	type XcmReserveTransferFilter = Everything;
+	type XcmTeleportFilter = Nothing;
+	type XcmReserveTransferFilter = Nothing;
```
Comment (apopiak) [inline polkadot-parachains/statemine/src/lib.rs:588]: This changes Statemine, should be Statemint. (same for #838

**Review by gavofyork — APPROVED** [review]

**Comment by gavofyork** [comment]
This is safe, but I don't know if we want to disable the reserve asset transfer or just the teleporting?

**Comment by KiChjang** [comment]
Hmm ok, if we disable reserve asset transfers, that means no one would be able to transfer reserve assets via the `reserve_transfer_assets` extrinsic on the XCM pallet. However, this does not mean `InitiateReserveWithdraw` XCMs are going to be disabled, since they're processed on the source chain first, and then subsequently a `WithdrawAsset` instruction is sent to Statemint and then executed, and the `XcmReserveTransferFilter` is not used.

All this is to say that disabling `XcmReserveTransferFilter` only affects whether or not users are able to dispatch `reserve_transfer_assets` on the XCM pallet on Statemint, and other XCM operations dealing with reserve assets are not affected. I can see that it can be a prudent step for us to take to set `XcmReserveTransferFilter` to `Nothing` if we are being conservative here and do not want to have any funny business going on with people attempting to exploit `reserve_transfer_assets`.

**Comment by joepetrowski** [comment]
Unless we change https://github.com/paritytech/polkadot/pull/4475 to allow reserve transfers to Statemint, I don't think it matters here as no DOT would arrive on Statemint (and thus asset creation isn't possible either because nobody could pay the deposit). I would rather restrict them so that opening them on the Relay Chain doesn't have the effect of making them bi-directional. That is, we allow teleports/reserve asset transfer on Statemint only after allowing them from the Relay Chain.

**Pull request merged**